### ISSUE TITLE
Crash :: libc++abi.dylib: Pure virtual function called!

### ISSUE THREAD

**Issue by Reggian** [realm/realm-cocoa#3121]
Hello again.

I have got another crash today. I was moving the app between foreground and background state.

```
libc++abi.dylib: Pure virtual function called!
```

<img width="1076" alt="line" src="https://cloud.githubusercontent.com/assets/1903894/12450593/06aed4fe-bf85-11e5-9c67-4dd59fb8b080.png">
<img width="1237" alt="stack" src="https://cloud.githubusercontent.com/assets/1903894/12450594/06d43f0a-bf85-11e5-9d24-848301544184.png">

```
Running on: iPhone 6s, iOS 9.2

ProductName:	Mac OS X
ProductVersion:	10.11.2
BuildVersion:	15C50

/Applications/Xcode.app/Contents/Developer
Xcode 7.2
Build version 7C68

/usr/local/bin/pod
0.39.0
Realm (0.97.0)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)

/usr/local/bin/carthage
0.10.0
(not in use here)

/usr/bin/git
git version 2.5.4 (Apple Git-61)
```

**Comment by tgoyne** [comment]
This also is probably fixed by https://github.com/realm/realm-cocoa/pull/3078.

**Comment by Reggian** [comment]
Thank you.
Will do and report if anything similar reoccurs.

**Comment by kishikawakatsumi** [comment]
I believe this issue is duplicate of #3013 and #3035. They are fixed at #3078. I'm going to close this issue. If you encounter the further problem, please feel free to reopen this issue or create a new one.
